haskell_scotty/ClientRepo.hs

```diff
@@ -16,3 +18,19 @@ getClients = do
 
 instance QueryResults Client where
   convertResults fs vs = Client $... zip fs vs
+
+saveClient :: Client -> IO Client
+saveClient client = do
+  conn <- connectDB
+  execute conn "INSERT INTO clients \
+    \(name,address,city,state,zip,country,email,contact,phone,created_at,updated_at) values \
+    \(?,?,?,?,?,?,?,?,?,?,?)" client
+  results <- query_ conn "SELECT LAST_INSERT_ID()"
+  let [Only clientId] = results
+  return client { Client.id = clientId }
+
+instance QueryParams Client where
+  renderParams
+    (Client _ name address city state zip country email contact phone created_at updated_at) =
+      [render name, render address, render city, render state, render zip, render country, render email, render contact, render phone, render created_at, render updated_at]
```

Comment:
The more I see of `mysql-simple` the more I'm not very happy with how they structured the interface. I might dig around for something with an ORM.